pypeit/core/telluric.py

```diff
@@ -320,58 +362,66 @@ def shift_telluric(tell_model, loglam, dloglam, shift, stretch):
+    and stretch.
 
     This routine performs the following steps:
 
-       1. nearest grid point interpolation of the telluric model onto a
-          new location in the 4-d space (pressure, temperature,
-          humidity, airmass)
+       1. summation of telluric PCA components multiplied by coefficients
+       
+       2. transformation of telluric PCA model from arsinh(tau) to transmission
 
-       2. convolution of the atmosphere model to the resolution set by
+       3. convolution of the atmosphere model to the resolution set by
           the spectral resolution.
 
-       3. (Optional) shift and stretch the telluric model.
+       4. shift and stretch the telluric model.
 
     Args:
         theta_tell (`numpy.ndarray`_):
-            Vector with the telluric model parameters. Must be 5 or 7
-            elements long. See method description.
+            Vector with tell_npca PCA coefficients (if teltype='PCA')
+            or pressure, temperature, humidity, and airmass (if teltype='grid'),
+            followed by spectral resolution, shift, and stretch.
+            Final length is then tell_npca+3 or 7.
         tell_dict (:obj:`dict`):
-            Dictionary containing the telluric grid data. See
-            :func:`read_telluric_grid`.
+            Dictionary containing the telluric data. See
+            :func:`read_telluric_pca` if teltype=='PCA'.
+            or
+            :func:`read_telluric_grid` if teltype=='grid'.
         ind_lower (:obj:`int`, optional):
             The index of the first pixel to include in the model. Selecting a
             wavelength region for the modeling makes things faster because we
             only need to convolve the portion that is needed for the current
             model fit.
-        ind_upper:
+        ind_upper (:obj:`int`, optional):
             The index (inclusive) of the last pixel to include in the model.
             Selecting a wavelength region for the modeling makes things
             faster because we only need to convolve the portion that is
             needed for the current model fit.
 
     Returns:
         `numpy.ndarray`_: Telluric model evaluated at the desired location
-        theta_tell in model atmosphere parameter space.
+        theta_tell in model atmosphere parameter space. Shape is given by
+        the size of `wave_grid' plus `tell_pad_pix' padding from the input
+        tell_dict.
+        
     """
     ntheta = len(theta_tell)
-    if ntheta not in [5, 7]:
-        msgs.error('Input model atmosphere parameters must have length 5 or 7.')
-
-    tellmodel_hires = interp_telluric_grid(theta_tell[:4], tell_dict)
-
+    teltype = tell_dict['teltype']
+    # FD: Currently assumes that shift and stretch are on.
+    # TODO: Make this work without shift and stretch.
+    if teltype == 'PCA':
+        ntell = ntheta-3
+    elif teltype == 'grid':
+        ntell = 4
+    else:
+        msgs.error("Unsupported teltype, must be 'PCA' or 'grid'")
```

Comment:
I think it's fine to catch this here, but I wonder if a better approach is to use the `validate()` method in `pypeitpar`, and then change the error message to read:
`msgs.error(f"Unsupported teltype: {teltype}")`
So that the options don't need to be updated here if there is ever a change to the allowed methods in `pypeitpar` (i.e. everything is then self-contained). In reality, this line should never be reached if it's in the `pypeitpar`, and the only reason it would be hit if a developer was adding a new option in the future.

Reply:
Removed.